### PR TITLE
Mobiledev

### PR DESCRIPTION
**Pull request by OAHC2022**
Please fill in the template below.

For your code:

-   [ ] Your changes are made on the dev branch or a new branch.
-   [ ] You have tested your code extensively.

For your pull request description:

-   [ ] Be clear that what type of changes you have made, e.g., fixed a bug, improved performance, changed/added functionality, updated documentation, etc.
-   [ ] Summarize what this pull request accomplishes and why this is worth doing.
-   [ ] (Optional) Include any of your design decisions and alternatives that you considered.
-   [ ] (Optional) Point out the part that you may want a more thorough review.
-   [ ] (Optional) Add any questions, comments, or concerns that you have, including those about your changes and those about the rest of this project.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -95,29 +95,63 @@
...
+            v-else-if="sideBar.showEvent"
+            class="bg-light"
+            :class="'sidebar-nocol' + classMobile"
+        >
+        </event-view>
+
+        <filter-view
+            v-else-if="sideBar.showFilter"
+            class="bg-light"
+            :class="'sidebar-nocol' + classMobile"
```
Comment (hanzhi713) [inline src/App.vue]: Don't use `sidebar-nocol` everywhere. It is a replacement of the `sidebar` class when you require the background of some `list-group-item` to be pure white. If it is not required, please do not use it.

File: src/components/tabs/ClassView.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import ClassList from '../ClassList.vue';
 
 /**
  * component for editing classes and manipulating schedules
- * @author Hanzhi Zhou
+ * @author Hanzhi Zhou, Zichao Hu
```
[the remark this answers is not in the document]
Comment (OAHC2022) [reply src/components/tabs/ClassView.ts]: Haha I just want to be there

File: src/components/tabs/EventView.vue [hunk 3]
```diff
@@ -105,7 +105,7 @@
                     v-for="event in schedule.currentSchedule.events"
                     :key="event.key"
                     class="mx-3 align-items-center justify-content-between "
-                    :class="event.selected && 'table-primary'"
+                    :class="event == currentSelectedEvent && 'table-primary'"
```
Comment (hanzhi713) [inline src/components/tabs/EventView.vue]: This is a hack of the logical shortcut...
A better way to write this is to use the Vue provided conditional class syntax:
```diff
- :class="event == currentSelectedEvent && 'table-primary'"
+ :class="{'table-primary': event === currentSelectedEvent}"
```
Btw, we use strict equality (`===`) whenever we can

File: src/components/tabs/EventView.ts [hunk 4]
```diff
@@ -15,11 +15,7 @@ import { to12hr, to24hr } from '@/utils';
 export default class EventView extends Store {
     get event() {
         // reset event.selected to false if it is not editing
-        if (!this.isEditingEvent) {
-            for (const ev of this.schedule.currentSchedule.events) {
-                ev.selected = false;
-            }
-        }
+        this.currentSelectedEvent = new Event("",false);
```
Comment (hanzhi713) [inline src/components/tabs/EventView.ts]: You can use nullable type here. That is, assign `currentSelectedEvent to null`.

**Comment by hanzhi713** [comment]
You also need to fix the CI build failure (will be very easy)

**Comment by hanzhi713** [comment]
Need further refactoring.

**Comment by hanzhi713** [comment]
Further refactoring done in
107199a5b224c7262f620236529cce25ddf8421d and
d24ccda27b2565182e71ca936de639fa7abd60d3

**Pull request merged**